src/BlockchainWriter/Router.ts

```diff
@@ -24,34 +24,45 @@ export class Router {
   }
 
   async start() {
-    await this.messaging.consume(Exchange.ClaimIPFSHash, this.onClaimIPFSHash)
+    await this.messaging.consume(Exchange.BatchWriterCreateNextBatchSuccess, this.onBatchWriterCreateNextBatchSuccess)
+    await this.messaging.consume(Exchange.BlockchainWriterTimestampRequest, this.onBlockchainWriterTimestampRequest)
   }
 
-  onClaimIPFSHash = async (message: any): Promise<void> => {
+  onBatchWriterCreateNextBatchSuccess = async (message: any): Promise<void> => {
+    const messageContent = message.content.toString()
+    const { fileHashes, directoryHash } = JSON.parse(messageContent)
+    if (fileHashes.length > 0)
+      this.messaging.publish(Exchange.BlockchainWriterTimestampRequest, { fileHashes, directoryHash })
+  }
+
+  onBlockchainWriterTimestampRequest = async (message: any): Promise<void> => {
     const logger = this.logger.child({ method: 'onClaimIPFSHash' })
 
     const messageContent = message.content.toString()
-    const { claimId, ipfsHash } = JSON.parse(messageContent)
+    const { fileHashes, directoryHash } = JSON.parse(messageContent)
 
     logger.trace(
       {
-        claimId,
-        ipfsHash,
+        fileHashes,
+        directoryHash,
       },
       'Timestamping requested'
     )
 
     try {
-      await this.claimController.requestTimestamp(ipfsHash)
+      await this.claimController.requestTimestamp(directoryHash)
+      this.messaging.publish(Exchange.BlockchainWriterTimestampSuccess, { fileHashes, directoryHash })
+      logger.trace('time stamping success', { fileHashes, directoryHash })
     } catch (exception) {
       logger.error(
         {
           exception,
-          claimId,
-          ipfsHash,
+          directoryHash,
+          fileHashes,
         },
         'Uncaught Exception while requesting timestamp'
       )
+      this.messaging.publish(Exchange.BlockchainWriterTimestampFailure, { error: exception, fileHashes, directoryHash })
```

Comment:
Always `await` promises even if you're not interested in the result, as they may `throw`.

Reply:
Because this promise is in the catch block, I believe even added await will not help. I need to reevaluate this code and see if I can solve this better.